plugins/woocommerce/src/Blocks/Shipping/ShippingController.php

```diff
@@ -488,4 +492,39 @@ function( $location ) {
 
 		return $served;
 	}
+
+	/**
+	 * Check if legacy local pickup is activated in any of the shipping zones or in the Rest of the World zone.
+	 *
+	 * @return bool
+	 */
+	public static function is_legacy_local_pickup_active() {
+		$has_legacy_pickup = false;
+
+		// Get all shipping zones.
+		$shipping_zones              = \WC_Shipping_Zones::get_zones( 'admin' );
+		$international_shipping_zone = new \WC_Shipping_Zone( 0 );
+
+		// Loop through each shipping zone.
+		foreach ( $shipping_zones as $shipping_zone ) {
+			// Get all registered rates for this shipping zone.
+			$shipping_methods = $shipping_zone['shipping_methods'];
+			// Loop through each registered rate.
+			foreach ( $shipping_methods as $shipping_method ) {
+				if ( 'local_pickup' === $shipping_method->id && 'yes' === $shipping_method->enabled ) {
+					$has_legacy_pickup = true;
+					break 2;
+				}
+			}
+		}
+
+		foreach ( $international_shipping_zone->get_shipping_methods( true ) as $shipping_method ) {
+			if ( 'local_pickup' === $shipping_method->id ) {
+				$has_legacy_pickup = true;
+				break;
+			}
+		}
+
+		return $has_legacy_pickup;
```

Comment:
Can we just return true and avoid breaking the foreach? Is there any point in continuing with the method? 🤔

So, replacing this with:

```suggestion
		// Get all shipping zones.
		$shipping_zones              = \WC_Shipping_Zones::get_zones( 'admin' );
		$international_shipping_zone = new \WC_Shipping_Zone( 0 );

		// Loop through each shipping zone.
		foreach ( $shipping_zones as $shipping_zone ) {
			// Get all registered rates for this shipping zone.
			$shipping_methods = $shipping_zone['shipping_methods'];
			// Loop through each registered rate.
			foreach ( $shipping_methods as $shipping_method ) {
				if ( 'local_pickup' === $shipping_method->id && 'yes' === $shipping_method->enabled ) {
					return true;
				}
			}
		}

		foreach ( $international_shipping_zone->get_shipping_methods( true ) as $shipping_method ) {
			if ( 'local_pickup' === $shipping_method->id ) {
				return true;
			}
		}

		return false;
```

Reply:
Thanks for pointing this out, @wavvves. I've applied @kloon's feedback from https://github.com/woocommerce/woocommerce/pull/45614#pullrequestreview-1938641357 and moved the updated method from `WC_Shipping` to `ShippingController`. This is the method I'm using now:

```php
public static function is_legacy_local_pickup_active() {
	$shipping_zones      = WC_Shipping_Zones::get_zones();
	$shipping_zones['0'] = array(
		'zone_name'        => __( 'Rest of the World', 'woocommerce' ),
		'shipping_methods' => WC_Shipping_Zones::get_zone( 0 )->get_shipping_methods(),
	);

	foreach ( $shipping_zones as $zone ) {
		foreach ( $zone['shipping_methods'] as $method ) {
			if ( 'local_pickup' === $method->id && $method->is_enabled() ) {
				return true;
			}
		}
	}

	return false;
}
```